test/metabase/util_test.cljc

```diff
@@ -470,11 +472,20 @@
     (is (= [:c :e] (u/conflicting-keys {:a 1 :b 2 :c 3 :e nil}
                                        {:b 2 :c 4 :d 5 :e 6})))))
 
+(defn- join [& crumbs] (str/join ":" crumbs))
+
 (deftest map-all-test
-  (let [join (fn [& crumbs] (str/join ":" crumbs))]
-    (testing "map-all with 1 collection is just map"
-      (is (= ["0" "1" "2" "3" "4"] (u/map-all join (range 5)))))
-    (testing "map-all works with 3 collections"
-      (is (= ["0:0" "1:1" "2:2" "3:" "4:"] (u/map-all join (range 5) (range 3)))))
-    (testing "map-all works with higher arity"
-      (is (= ["0:0:0" "1:1:1" "2:2:2" "3::3" "::4"] (u/map-all join (range 4) (range 3) (range 5)))))))
+  (testing "map-all with 1 collection is just map"
+    (is (= ["0" "1" "2" "3" "4"] (u/map-all join (range 5)))))
+  (testing "map-all works with 3 collections"
+    (is (= ["0:0" "1:1" "2:2" "3:" "4:"] (u/map-all join (range 5) (range 3)))))
+  (testing "map-all works with higher arity"
+    (is (= ["0:0:0" "1:1:1" "2:2:2" "3::3" "::4"] (u/map-all join (range 4) (range 3) (range 5))))))
+
+#?(:clj
+   (defspec map-all-returns-max-coll-input-size-test 1000
+     (prop/for-all [xs (mg/generator [:sequential :int])
```

Comment:
How simple is it to extend this to `n pos?` collections?

Reply:
It is not hard, but it's a bit harder to read:

```clojure
(defspec map-all-returns-max-coll-input-size-test 100
  (prop/for-all [xss (mg/generator [:sequential {:min 1} ;; <-- map-all requires at least 1 collection
                                    [:sequential pos?]])]
    (let [expected (apply
                    #(apply (partial u/map-all vector) %&)
                    xss)]
      (= (count expected)
         (apply max (map count xss))))))
```

Reply:
for the `pos?` bit, we could use :any, but that can sometimes create very large examples. So I've found that scalar values work best

Reply:
Oh, by `pos?` I just meant that `{:min 1}` part. Sticking with `:int` is good.